### ISSUE TITLE
Literal infers tuple literal as Union of the items though this syntax is forbidden by PEP 586

### ISSUE THREAD

**Issue by Dominik1123** [python/mypy#7783]
The section [Illegal parameters for `Literal` at type check time](https://www.python.org/dev/peps/pep-0586/#id25) describes that tuple literals inside `Literal` are forbidden:

> Tuples containing valid literal types like `Literal[(1, "foo", "bar")]`. The user could always express this type as `Tuple[Literal[1], Literal["foo"], Literal["bar"]]` instead. Also, tuples are likely to be confused with the `Literal[1, 2, 3]` shortcut.

Nevertheless mypy seems to be taking this shortcut since it doesn't flag the following code:

    from typing_extensions import Literal

    a : Literal[(1, 2)] = 1 

Python 3.7, mypy 0.740, typing-extensions 3.7.4.

**Comment by JelleZijlstra** [comment]
This is because the AST is the same in both cases, so mypy currently has no way of telling that these parentheses are there.

**Comment by Dominik1123** [comment]
I see. Can't you use the `lineno` and `col_offset` attributes to take a closer look?

Or maybe even better, change the specs, such that `Literal[1, 2]` and `Literal[(1, 2)]` are equivalent. Then one could also allow `Literal[x]` where `x` can be resolved to a literal tuple containing only literals. E.g.:

    x = (1, 2)
    def foo(y: Literal[x]): pass

This would have the benefit that one could use `x` also at runtime to perform checks on the values, such as `assert y in x`.

Because as the PEP mentions,if someone wants to declare a tuple literal they can always do so via `Tuple[Literal[1], Literal[2]]`.

**Comment by ilevkivskyi** [comment]
I don't think we should play any tricks with the parser. Instead, I would propose to clarify the text in the PEP mentioning that `Literal[(x, y)]` will be interpreted as `Literal[x, y]` because these two are equivalent in Python grammar.

cc @Michael0x2a
